### DISCUSSION TITLE
Can a menu item's icon be changed at runtime?

### DISCUSSION THREAD

**Discussion by reduckted** [VsixCommunity/Community.VisualStudio.Toolkit#212] [Q&A] [answered]
A menu item's icon can be set in the `.vsct` file by setting the `<Icon>` element.

```xml
<Button guid="MyPackage" id="MyCommand" priority="0x0100" type="Button">
    <Icon guid="ImageCatalogGuid" id="SettingsFile" />
    <CommandFlag>IconIsMoniker</CommandFlag>
</Button>
```

Does anyone know if the same thing can be done via code at runtime? 

I'm creating a [dynamic command](https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/blob/f561cc80d69ec8a177c9eecbbbb751cfebf0d9f7/src/toolkit/Community.VisualStudio.Toolkit.Shared/Commands/BaseDynamicCommand.cs), and I'd like to give each of the generated menu items a different icon, but the icon won't be known until runtime. The menu item text can be changed via the `OleMenuCommand.Text` property, but there doesn't seem to be any properties that allow the icon to be changed.

**Comment by madskristensen** [comment]
I haven't figured out how to do it. I'm not sure it's supported because of the somewhat static nature of the VS command table. @ryanmolden might know?

**Comment by reduckted** [comment, marked as answer]
From here: https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/discussions/272#discussioncomment-2008462

The answer is no. 😞
